### PR TITLE
Document integration events for Charges and Charge Prices

### PR DESCRIPTION
**Pull request by prtandrup**
## Description

This PR documents the **charges** and **charge prices** integration events targeting the needs of the Aggregations Domain.

Note: Creating the `IntegrationEventContract.proto` file is to follow in a subsequent story, #372  

## References

* #352

### CODE REVIEW

File: docs/contracts.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+<a name=".ChargeCreated"></a>
+
+### ChargeCreated
+
+Represents the creation of a new charge.
+This event contains the static charge data as well the initial charge period data.
+
+| Field | Type | Label | Description |
+| ----- | ---- | ----- | ----------- |
+| ChargeId | string | required | A charge identifier provided by the Market Participant. Combined with Charge Owner it becomes unique  |
```
Comment (bemwen) [inline docs/contracts.md:31]: Combined with owner and type

File: docs/contracts.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+| ChargeType | enum | required | The type of charge; tariff, fee or subscription |
+| ChargeOwner | string | required | A charge owner identification, e.g. the Market Participant's GLN or EIC number |
+| Currency | string | required | The charge price currency |
+| Resolution | enum | required | The resolution of a charge price list, e.g. 15 min, hourly, daily, monthly |
+| TaxIndicator | bit | required | Indicates whether a tariff is considered a tax or not, 1 = true |
+| StartDateTime | Timestamp (UTC) | required | The charge period's valid from date and time |
+| EndDateTime | Timestamp (UTC), nullable | optional | The charge period's valid to date and time |
```
Comment (bemwen) [inline docs/contracts.md:38]: It might be misleading to have a period on the charge data which cannot change over time?

File: docs/contracts.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+<a name=".ChargePeriodsUpdated"></a>
+
+### ChargePeriodsUpdated
+
+Represents the update of charge periods details.
+This event contains the new period as well other periods that were adjusted by it.
+
+| Field | Type | Label | Description |
+| ----- | ---- | ----- | ----------- |
+| ChargeId | string | required | A charge identifier provided by the Market Participant. Combined with Charge Owner it becomes unique  |
```
Comment (bemwen) [inline docs/contracts.md:50]: For uniqueness, include type as well

File: docs/contracts.md (hunk 1 above)
Comment (prtandrup) [reply docs/contracts.md:31]: forgotten, thx. fixed.

File: docs/contracts.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+| ChargeType | enum | required | The type of charge; tariff, fee or subscription |
+| ChargeOwner | string | required | A charge owner identification, e.g. the Market Participant's GLN or EIC number |
+| CompleteUpdatePeriodStartDateTime | Timestamp (UTC) | required | The starting point of the earliest charge period reported in this event |
```
Comment (bemwen) [inline docs/contracts.md:53]: Maybe we can get away with the shorter name UpdatedPeriodStartDateTime?

File: docs/contracts.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+<a name=".ChargePeriodsUpdated.ChargePeriod"></a>
+
+#### ChargePeriodsUpdated.ChargePeriod
+
+Represents a charge period
+
+| Field | Type | Label | Description |
+| ----- | ---- | ----- | ----------- |
+| StartDateTime | Timestamp (UTC) | required | The charge period's valid from date and time |
+| EndDateTime | Timestamp (UTC), nullable | optional | The charge period's valid to date and time |
```
Comment (bemwen) [inline docs/contracts.md:66]: No other fields required?

File: docs/contracts.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+| Field | Type | Label | Description |
+| ----- | ---- | ----- | ----------- |
+| ChargeId | string | required | A charge identifier provided by the Market Participant. Combined with Charge Owner it becomes unique  |
+| ChargeType | enum | required | The type of charge; tariff, fee or subscription |
+| ChargeOwner | string | required | A charge owner identification, e.g. the Market Participant's GLN or EIC number |
+| EffectiveDate | Timestamp (UTC) | required | The date and time the charge is discontinued |
+
+<a name=".ChargeDiscontinuationCancelled"></a>
+
+### ChargeDiscontinuationCancelled
```
Comment (bemwen) [inline docs/contracts.md:83]: For human errors?

File: docs/contracts.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+### ChargePricesUpdated
+
+Represents the update of one or more charge prices.
+
+| Field | Type | Label | Description |
+| ----- | ---- | ----- | ----------- |
+| ChargeId | string | required | A charge identifier provided by the Market Participant. Combined with Charge Owner it becomes unique  |
+| ChargeType | enum | required | The type of charge; tariff, fee or subscription |
+| ChargeOwner | string | required | A charge owner identification, e.g. the Market Participant's GLN or EIC number |
+| PriceListStartDateTime | Timestamp (UTC) | required | The time interval covers the entire period of charge prices, the starting point here is equal to the time of the very first charge price in the list  |
```
Comment (bemwen) [inline docs/contracts.md:105]: Might be confusing to introduce PriceLists at this point? Maybe just UpdatePeriodStartDateTime again?

File: docs/contracts.md (hunk 2 above)
Comment (prtandrup) [reply docs/contracts.md:38]: Given our talk I will separate the charge period dates from the ChargeCreated event and keep only the static charge data. I will reconsider whether there is a need to publish charge period dates, when no dynamic charge period data is published.

File: docs/contracts.md (hunk 5 above)
Comment (prtandrup) [reply docs/contracts.md:66]: No other fields required by the Aggregation domain as we know it.

File: docs/contracts.md (hunk 6 above)
Comment (prtandrup) [reply docs/contracts.md:83]: Yep, for correcting human errors.

File: docs/contracts.md (hunk 4 above)
Comment (prtandrup) [reply docs/contracts.md:53]: Applied, thx.

File: docs/contracts.md (hunk 7 above)
Comment (prtandrup) [reply docs/contracts.md]: Done.

**Review by Mech0z — APPROVED** [review]

**Review by lasrinnil — APPROVED** [review]

**Pull request merged**